pkg/terraformer/terraformer.go

```diff
@@ -290,15 +314,14 @@ func (t *Terraformer) updateObjects(ctx context.Context, log logr.Logger, patchO
 	return err
 }
 
-func (t *Terraformer) updateObjectFinalizers(ctx context.Context, log logr.Logger, obj controllerutil.Object, patchObj func(controllerutil.Object, string)) error {
-	key, err := client.ObjectKeyFromObject(obj)
-	if err != nil {
-		log.Error(err, "failed to construct key", "object", obj)
-		return err
-	}
+func (t *Terraformer) updateObjectFinalizers(ctx context.Context, log logr.Logger, obj client.Object, patchObj func(client.Object, string)) error {
+	var (
+		key = client.ObjectKeyFromObject(obj)
+		err error
+	)
 
 	for i := 0; i < maxPatchRetries; i++ {
-		err = t.client.Get(ctx, key, obj)
+		err := t.client.Get(ctx, key, obj)
```

Comment:
This hides the `err` var in the outer scope, see https://play.golang.org/p/HGtiLuPr-Iv for example.
```suggestion
		err = t.client.Get(ctx, key, obj)
```